### PR TITLE
Watcher removal

### LINKED ISSUE #1891: Ask if removal as maintainer should also remove them as watcher
This is based https://github.com/FAIRsharing/FAIRsharing-API/issues/604 - please see the description there for details.
The PR linked to that ticket covers allowing curators to send an array of watcher_ids; this ticket should cover modifying the GeneralInfo section of the editor to process those ids and send them to the server.

### PR DESCRIPTION
**Pull request by knirirr**
This deals with https://github.com/FAIRsharing/fairsharing.github.io/issues/1891 (which also requires https://github.com/FAIRsharing/FAIRsharing-API/pull/610).

### CODE REVIEW

**Comment by knirirr** [comment]
The necessary server-side code to try this out is not currently deployed; it will have to wait until after https://github.com/FAIRsharing/fairsharing.github.io/pull/1890 is dealt with.

**Comment by knirirr** [comment]
This can now be viewed as the correct server code is available.

**Review by ramGranell — COMMENTED** [review]
Neither @prakhyatox  or myself can go to any record page in the Netlify (it appears 404 error), e.g. https://deploy-preview-1892--fairsharing.netlify.app/FAIRsharing.m9x0zg . I don't know if it is a Netlify issue or the error is in other place.

**Comment by knirirr** [comment]
>  I don't know if it is a Netlify issue or the error is in other place.

If this sort of thing happens then the reason can usually be seen in the network console. In this case it's the API being unhappy about missing fields etc.

**Review by ramGranell — COMMENTED** [review]
Now, I can access to the records. I'm not sure if I'm testing it properly but removing myself as a maintainer of a record does not ask to be removed as a watcher. I did the following steps: 1) I claimed to be maintainer of a record option that appears in the record menu) , 2) I approved in the curator panel to be the maintainer, 3) I became watcher of a record option that appears in the record menu) , 4) I stop being the maintainer of the record (option that appears in the record menu) and 5) I keep being the watcher of the record but not the maintainer.

**Comment by knirirr** [comment]
Step (4) is the problem here - you will need to remove yourself as a maintainer from the general info. section of the record page.

**Review by ramGranell — APPROVED** [review]
It seems to work properly when removing maintainer from the edit menu of the record "General Information".

File: src/components/Editor/GeneralInformation/Maintainers.vue [hunk 1]
```diff
@@ -170,7 +212,25 @@ export default {
       };
     },
     removeMaintainer(maintainerIndex){
+      let _module = this;
+      let maintainerId = this.maintainers[maintainerIndex].id;
       this.maintainers.splice(maintainerIndex, maintainerIndex+1)
```
Comment (prakhyatox) [inline src/components/Editor/GeneralInformation/Maintainers.vue]:
```suggestion
      _module.maintainers.splice(maintainerIndex, maintainerIndex+1)
```

File: src/components/Editor/GeneralInformation/Maintainers.vue [hunk 2]
```diff
@@ -170,7 +212,25 @@ export default {
...
+      if (_module.watchers.some(m => m.id === maintainerId)) {
+        this.watcherToRemove = maintainerId;
```
Comment (prakhyatox) [inline src/components/Editor/GeneralInformation/Maintainers.vue]:
```suggestion
        _module.watcherToRemove = maintainerId;
```

File: src/components/Editor/GeneralInformation/Maintainers.vue [hunk 3]
```diff
@@ -170,7 +212,25 @@ export default {
...
+        this.showRemoveWatcher = true;
```
Comment (prakhyatox) [inline src/components/Editor/GeneralInformation/Maintainers.vue]:
```suggestion
        _module.showRemoveWatcher = true;
```

File: src/components/Editor/GeneralInformation/Maintainers.vue [hunk 4]
```diff
@@ -170,7 +212,25 @@ export default {
...
+      let _module = this;
+      if (removeWatcher) {
+        let index = _module.watchers.findIndex( element => {
+          if (element.id === _module.watcherToRemove) {
+            return true;
+          }
+        });
+        _module.watchers.splice(index, index+1);
+      }
+      this.showRemoveWatcher = false;
```
Comment (prakhyatox) [inline src/components/Editor/GeneralInformation/Maintainers.vue]:
```suggestion
      _module.showRemoveWatcher = false;
```

File: src/components/Editor/GeneralInformation/Maintainers.vue [hunk 5]
```diff
@@ -214,10 +214,10 @@ export default {
     removeMaintainer(maintainerIndex){
       let _module = this;
       let maintainerId = this.maintainers[maintainerIndex].id;
```
Comment (prakhyatox) [inline src/components/Editor/GeneralInformation/Maintainers.vue]:
```suggestion
      let maintainerId = _module.maintainers[maintainerIndex].id;
```

**Review by prakhyatox — APPROVED** [review]

**Review by allysonlister — APPROVED** [review]
Works as expected, thank you!

**Pull request merged**